syntax = "proto3";

package fetch.aea.fipa;

message FIPAMessage{

    message CFP{
        message Nothing {
        }
        oneof query{
            bytes bytes = 1;
            Nothing nothing = 2;
            bytes query_bytes = 3;
        }
    }
    message Propose{
        repeated bytes proposal = 1;
    }
    message Accept{}
    message MatchAccept{}
    message Decline{}
    message Inform{
        bytes bytes = 1;
    }

     message Accept_W_Address{
        string address = 1;
    }

    message MatchAccept_W_Address{
        string address = 1;
    }

    int32 message_id = 1;
    int32 dialogue_id = 2;
    int32 target = 3;
    oneof performative{
        CFP cfp = 4;
        Propose propose = 5;
        Accept accept = 6;
        MatchAccept match_accept = 7;
        Decline decline = 8;
        Inform inform = 9;
        Accept_W_Address accept_w_address = 10;
        MatchAccept_W_Address match_accept_w_address = 11;
    }
}
